test/Microsoft.AspNetCore.Mvc.Localization.Test/Internal/MvcLocalizationServiceCollectionExtensionsTest.cs

```diff
@@ -283,61 +116,102 @@ public void AddLocalizationServicesWithLocalizationOptions_AddsNeededServices()
-                    Assert.Equal(typeof(IOptionsSnapshot<>), service.ServiceType);
-                    Assert.Equal(typeof(OptionsManager<>), service.ImplementationType);
-                    Assert.Equal(ServiceLifetime.Scoped, service.Lifetime);
-                },
-                service =>
-                {
-                    Assert.Equal(typeof(IOptionsMonitor<>), service.ServiceType);
-                    Assert.Equal(typeof(OptionsMonitor<>), service.ImplementationType);
-                    Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
-                },
-                service =>
-                {
-                    Assert.Equal(typeof(IOptionsFactory<>), service.ServiceType);
-                    Assert.Equal(typeof(OptionsFactory<>), service.ImplementationType);
-                    Assert.Equal(ServiceLifetime.Transient, service.Lifetime);
-                },
-                service => 
-                {
-                    Assert.Equal(typeof(IOptionsMonitorCache<>), service.ServiceType);
-                    Assert.Equal(typeof(OptionsCache<>), service.ImplementationType);
-                    Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
-                },
+            AssertContainsExpected(collection, ExpectedServices);
+        }
 
-                service =>
-                {
-                    Assert.Equal(typeof(IConfigureOptions<RazorViewEngineOptions>), service.ServiceType);
-                    Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
-                },
-                service =>
+        private static void AssertContainsExpected(IServiceCollection services, IDictionary<Type, Tuple<object, ServiceLifetime>> expectedServices)
+        {
+            foreach(var expectedService in expectedServices)
+            {
+                var implementationType = expectedService.Value.Item1 as Type;
+                var expectedCount = expectedService.Value.Item1 as int?;
+                if(implementationType != null)
                 {
-                    Assert.Equal(typeof(IHtmlLocalizerFactory), service.ServiceType);
-                    Assert.Equal(typeof(HtmlLocalizerFactory), service.ImplementationType);
-                    Assert.Equal(ServiceLifetime.Singleton, service.Lifetime);
-                },
-                service =>
+                    AssertContainsSingleType(services, expectedService.Key, implementationType, expectedService.Value.Item2);
+                }
+                else if(expectedCount != null)
                 {
-                    Assert.Equal(typeof(IHtmlLocalizer<>), service.ServiceType);
-                    Assert.Equal(typeof(HtmlLocalizer<>), service.ImplementationType);
-                    Assert.Equal(ServiceLifetime.Transient, service.Lifetime);
-                },
-                service =>
+                    AssertContainsMultiple(services, expectedService.Key, expectedCount.Value, expectedService.Value.Item2);
```

Comment:
We don't have any multi registrations \ instances here. Can we just remove these for now?